src/vec.rs

```diff
@@ -35,34 +38,433 @@ pub trait FallibleVec<T> {
     fn try_insert(&mut self, index: usize, element: T) -> Result<(), (T, TryReserveError)>;
     /// see append
     fn try_append(&mut self, other: &mut Self) -> Result<(), TryReserveError>;
-    /// see resize, only works when the `value` implements Copy, otherwise, look at try_resize_no_clone
+    /// see resize, only works when the `value` implements Copy, otherwise, look at try_resize_no_copy
     fn try_resize(&mut self, new_len: usize, value: T) -> Result<(), TryReserveError>
     where
         T: Copy + Clone;
+    fn try_resize_with<F>(&mut self, new_len: usize, f: F) -> Result<(), TryReserveError>
+    where
+        F: FnMut() -> T;
     /// resize the vec by trying to clone the value repeatingly
     fn try_resize_no_copy(&mut self, new_len: usize, value: T) -> Result<(), TryReserveError>
     where
         T: TryClone;
     /// see resize, only works when the `value` implements Copy, otherwise, look at try_extend_from_slice_no_copy
     fn try_extend_from_slice(&mut self, other: &[T]) -> Result<(), TryReserveError>
     where
-        T: Copy + Clone;
+        T: Clone;
```

Comment:
here the try_extend_from_slice function can panic on OOM if T is not copy and clone() fails
(because I have used `extend_from_slices` which "Iterates over the slice other, clones each element, and then appends it to this Vec").

Reply:
Ah, good point. Will revert that bit.